specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2021-09-01-privatepreview/managedcluster.json

```diff
@@ -0,0 +1,1561 @@
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/managedClusterVersions": {
+      "get": {
+        "operationId": "ManagedClusterVersion_List",
+        "summary": "Gets the list of Service Fabric cluster code versions available for the specified location.",
+        "description": "Gets all available code versions for Service Fabric cluster resources by location.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/location"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "x-ms-examples": {
+          "List cluster versions": {
+            "$ref": "./examples/ManagedClusterVersionList_example.json"
+          }
+        },
+        "tags": [
+          "ManagedClusterVersion"
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ManagedClusterCodeVersionListResult"
+            }
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/environments/{environment}/managedClusterVersions": {
+      "get": {
+        "operationId": "ManagedClusterVersion_ListByEnvironment",
+        "summary": "Gets the list of Service Fabric cluster code versions available for the specified environment.",
+        "description": "Gets all available code versions for Service Fabric cluster resources by environment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/location"
+          },
+          {
+            "$ref": "#/parameters/environment"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "x-ms-examples": {
+          "List cluster versions by environment": {
+            "$ref": "./examples/ManagedClusterVersionListByEnvironment.json"
+          }
+        },
+        "tags": [
+          "ManagedClusterVersion"
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ManagedClusterCodeVersionListResult"
+            }
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/managedUnsupportedVMSizes": {
```

Comment:
@a-santamaria I diffed your 1st & 2nd iteration, I see this is the API that got added in new version. This API is GET collection API. You need to add GET individual resource too. 
Any resource type MUST implement GET for collection and single resource regardless if it's a singleton or not. 
For example GET /subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/managedUnsupportedVMSizes/{vmSize}

Please refer best practice 'Inconsistent or incomplete CRUD' here https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html

Reply:
@RamyasreeChakka added the new path. Please take a look again

Reply:
@a-santamaria - Signed off from ARM side. Please work with your PR assignee to merge the PR.